flatten/merge_allof.go

```diff
@@ -1057,3 +1057,13 @@ func resolveDefault(collection *SchemaCollection) (interface{}, error) {
 	}
 	return first, nil
 }
+
+func firstNonEmptyString(values []string) string {
+	if len(values) == 1 {
+		return values[0]
+	}
+	if len(values[0]) > 0 {
```

Comment:
This condition len(values)>0 includes the prevision one: len(values)==1
Why two tests?